scripts/gendecoder/main.go

```diff
@@ -95,6 +95,8 @@ func handleBasic(getter *getter, name, kind string) {
 	switch kind {
 	case "int", "int8", "int16", "int32", "int64", "uint", "uint8", "uint16", "uint32", "uint64":
 		getter.Fields = append(getter.Fields, field{Name: name, Type: "int64"})
+	case "float32", "float64":
```

Comment:
unfortunately this is not enough to have float handled by the gendecoder. The function used to retrieve the value are only generated for int64 and I think that the gremlin traversal thing will have to be updated too. Any chance to have int64 only values ? conversion into ms ?

Reply:
Ok.  I changed to int64 - it looks like it is just truncating the floating point. I think for a first version that will be ok.

I removed the "LastUpdate" code.  It doesn't really make sense for rates (like KB per second).  I think we can come back to that at some point.

I'd like to be able to display some IO rates on the edges - but it looks like a fair amount of work for the legacy UI.  So, if you support it in the new UI - please keep in mind that a block device link would want to display the rates as IOPs or KB/s etc.

Does the way I implemented the "metrics" make sense?  I didn't see a better way.  The two structures are necessary for parsing the output of iostat and the other for passing to the UI. (?)

Reply:
Thanks for changing to int64, we discussed about this limitation during our last public weekly call, we will remove this limitation in the future. I plan to work soon on the link labels on the "new" UI, I'll take care of this. Having two structures one for parsing and another one for the output seems to be a good approach (IMHO).